src/helpers/get-wallets-from-provider.tsx

```diff
@@ -27,15 +29,28 @@ export async function* getWalletsFromProvider(
   })[] = [];
 
   const genHdPath = (_index: number) => {
+    if (provider instanceof ProviderKeystoneReactNative) {
+      return provider.buildPath(_index);
+    }
     if (mnemonicType === ChooseAccountTabNames.Basic) {
       return `${ETH_HD_SHORT_PATH}/${_index}`;
     }
     return LEDGER_HD_PATH_TEMPLATE.replace('index', String(_index));
   };
 
   while (canNext) {
+    const keystoneWalletsCount =
+      Wallet.getAll().filter(w => w.type === WalletType.keystone)?.length || 0;
+    const keystoneName = getText(I18N.keystoneWalletAccountNumber, {
+      number: `${keystoneWalletsCount + index + 1}`,
```

Comment:
```suggestion
      number: `${keystoneWalletsCount + 1}`,
```

Reply:
Wallet is mobx store so this function should react to keystoneWalletsCount. Or you can get the implementation from Ledger